scripts/zones/The_Eldieme_Necropolis/globals.lua

```diff
@@ -4,57 +4,61 @@
-    }
-
-    local tradeCount = GetNPCByID(ID.npc.CANDLE_OFFSET):getLocalVar('SkullTradeCount')  -- Track how many candles have been lit
-    local tradeWindow = GetNPCByID(ID.npc.CANDLE_OFFSET):getLocalVar('SkullTradeTimer') -- Track how much time before candles reset
-    local active = npc:getLocalVar('candleActive')                                      -- Track if current candle has already been lit
-
-    for i = 1, 5 do
-        if tradeCount == 6 and os.time() < tradeWindow and os.time() > active then         -- Final candle, spawn Skulls
-            GetNPCByID(ID.npc.CANDLE_OFFSET):setLocalVar('SkullTradeCount', 0)
-            GetNPCByID(ID.npc.CANDLE_OFFSET):setLocalVar('SkullRespawn', os.time() + 3600) -- 1 hour cooldown to respawn skulls
-            player:messageSpecial(ID.text.SKULL_SPAWN)
-            player:confirmTrade()
-
-            -- Spawn all 7 Skulls
-            for skull = 1, 7 do
-                SpawnMob(ID.mob.LICH_C_MAGNUS + skull) -- IDs based off Lich C Magnus
-            end
+    -- Used to stop Skulls from being spawned until an hour from being spawned
+    SetServerVariable('[ELDIEME]TimeToRespawnSkulls', os.time() + 3600)
+end
+
+local brazierMessages =
+{
+    ID.text.SKULL_SPAWN,
+    ID.text.SKULL_ONE_REMAIN,
+    ID.text.SKULL_TWO_REMAIN,
+    ID.text.SKULL_THREE_REMAIN,
+    ID.text.SKULL_FOUR_REMAIN,
+    ID.text.SKULL_FIVE_REMAIN,
+    ID.text.SKULL_SIX_REMAIN,
+}
 
-            break
-        elseif tradeCount == i and os.time() < tradeWindow and os.time() > active then -- Candle trades 2 through 6
-            GetNPCByID(ID.npc.CANDLE_OFFSET):setLocalVar('SkullTradeCount', i + 1)
-            npc:setLocalVar('candleActive', os.time() + 10)
-            player:messageSpecial(ID.text.THE_BRAZIER_IS_LIT)
-            player:messageSpecial(candleCount[i])
-            player:confirmTrade()
-            break
-        elseif os.time() > tradeWindow and os.time() > active then -- First candle trade to start timer
-            GetNPCByID(ID.npc.CANDLE_OFFSET):setLocalVar('SkullTradeCount', 1)
-            GetNPCByID(ID.npc.CANDLE_OFFSET):setLocalVar('SkullTradeTimer', os.time() + 40)
-            npc:setLocalVar('candleActive', os.time() + 10)
-            player:messageSpecial(ID.text.THE_BRAZIER_IS_LIT)
-            player:messageSpecial(ID.text.SKULL_SIX_REMAIN)
-            player:confirmTrade()
-            break
+local countUnlitBraziers = function()
+    local numberNotLit = 7
+
+    for i = 0, 6 do
+        local brazier = GetNPCByID(ID.npc.CANDLE_OFFSET + i)
+        if brazier and brazier:getAnimation() == xi.anim.OPEN_DOOR then
+            numberNotLit = numberNotLit - 1
         end
     end
+
+    return numberNotLit
 end
 
-local eldiemeGlobal =
-{
+local lightBrazier = function(player, npc)
+    npc:setAnimation(xi.anim.OPEN_DOOR)
+
+    --unlight brazier after five minutes
+    npc:timer(300000, function()
+        npc:setAnimation(xi.anim.CLOSE_DOOR)
+    end)
```

Comment:
Just so you are aware in the future, there is literally a function specifically to open a door, wait a specified number of seconds, and then close the same door, performing all 3 of these functions with one invocation without any seconds to milliseconds conversion or potential snags you can hit by invoking `timer()` directly. There is also a matching function to do a the reverse (closing a default open door, and then reopening it after a timeout).

`npc:openDoor(300)` would have done exactly what you did here in a single line.